File: web/src/types/calendarImport.ts
export interface CalendarPreview {
  uid: number;
  name: string;
  is_base: boolean;
  working_days: number[];
  hours_per_day: number;
  holidays: number;
}

export interface ResourceMapping {
  ms_project_resource: string;
  matched_resource_id: string | null;
  matched_resource_name: string | null;
  calendar_name: string;
}

export interface CalendarImportPreviewResponse {
  calendars: CalendarPreview[];
  resource_mappings: ResourceMapping[];
  total_holidays: number;
  date_range_start: string;
  date_range_end: string;
  warnings: string[];
}

export interface CalendarImportResponse {
  success: boolean;
  resources_updated: number;
  calendar_entries_created: number;
  templates_created: number;
  warnings: string[];
}

File: api/src/schemas/calendar_import.py
"""Pydantic schemas for calendar import."""

from datetime import date
from typing import Any
from uuid import UUID

from pydantic import BaseModel


class CalendarImportPreviewResponse(BaseModel):
    """Response schema for calendar import preview."""

    calendars: list[dict[str, Any]]
    resource_mappings: list[dict[str, Any]]
    total_holidays: int
    date_range_start: date
    date_range_end: date
    warnings: list[str]


class CalendarImportRequest(BaseModel):
    """Request schema for calendar import."""

    program_id: UUID
    start_date: date
    end_date: date
    resource_mapping: dict[str, UUID] | None = None


class CalendarImportResponse(BaseModel):
    """Response schema for calendar import."""

    success: bool
    resources_updated: int
    calendar_entries_created: int
    templates_created: int
    warnings: list[str]


class CalendarTemplateListResponse(BaseModel):
    """Response schema for listing calendar templates."""

    templates: list[dict[str, Any]]
    total: int
